controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,233 @@
+		if err := ctrl.SetControllerReference(sc, odfInfoConfigMap, r.Scheme); err != nil {
+			return err
+		}
+		r.Log.Info("Creating or updating odf-info configmap", OdfInfoMapKind, klog.KRef(sc.Namespace, OdfInfoConfigMapName))
+		obj.mutex.Lock()
+		odfInfoData, configErr := getOdfInfoData(r, sc)
+		if configErr != nil {
+			return fmt.Errorf("failed to get ODF info config data: %w", configErr)
+		}
+		odfInfoKey := obj.getOdfInfoKeyName(sc)
+		// Creates or appends to the data map
+		odfInfoConfigMap.Data[odfInfoKey] = odfInfoData
+		obj.mutex.Unlock()
+		return nil
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update odf-info config", OdfInfoMapKind, klog.KRef(sc.Namespace, OdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update odf-info config: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *odfInfoConfig) getOdfInfoKeyName(sc *ocsv1.StorageCluster) string {
+	return sc.Namespace + "/" + sc.Name + "." + OdfInfoKeyName
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	var odfInfoConfigMap corev1.ConfigMap
+	if err = r.Client.Get(r.ctx, types.NamespacedName{Namespace: operatorNamespace}, &odfInfoConfigMap); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if len(odfInfoConfigMap.Data) > 1 {
+		obj.mutex.Lock()
+		delete(odfInfoConfigMap.Data, obj.getOdfInfoKeyName(sc))
+		obj.mutex.Unlock()
+	} else {
+		obj.mutex.Lock()
+		if err = r.Client.Delete(r.ctx, &odfInfoConfigMap); err != nil {
+			r.Log.Error(err, "Failed to delete odf-info.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+		obj.mutex.Unlock()
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (string, error) {
+	var odfVersion, cephFSId string
+	var err error
+	if odfVersion, err = getOdfVersion(r, sc); err != nil {
+		return "", err
+	}
+	if cephFSId, err = getCephFsid(r, sc); err != nil {
+		return "", err
+	}
+	var odfDeploymentType string
+	if sc.Spec.ExternalStorage.Enable {
+		odfDeploymentType = OdfDeploymentTypeExternal
+	} else {
+		odfDeploymentType = OdfDeploymentTypeInternal
+	}
+	var isDROptimized = false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !sc.Spec.ExternalStorage.Enable {
+		if isDROptimized, err = getIsDROptimized(r, sc); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(sc); err != nil {
+		return "", err
+	}
```

Comment:
Purposely declared separately as otherwise won't be recognized in `data`:
```go
	var storageSystemName string
	if storageSystemName, err = getStorageSystemName(sc); err != nil {
		return "", err
	}
	storageClusterCount := len(r.clusters.GetStorageClusters())

	data := OdfInfoData{
		StorageSystemName:   storageSystemName,
```